File: LeetCode-Cpp/MergeStringsAlternately.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    string mergeAlternately(string word1, string word2) {
         std::string merged;
    size_t length = std::max(word1.length(), word2.length());

    for (size_t i = 0; i < length; ++i) {
        if (i < word1.length()) {
            merged += word1[i];
        }
        if (i < word2.length()) {
            merged += word2[i];
        }
    }

    return merged;
    }
   
};

//int main(int argc, char **argv)
//{
//    Solution solution;
  //  cout << solution.mergeAlternately("abcdc", "pqrolm");
    //return 0;
//}

File: LeetCode-Python/MergeStringsAlternately.py
class Solution:

    def mergeAlternately(word1, word2):

        length = max(len(word1),len(word2))
        merged = ""

        for x in range(length):
            if len(word1)>len(word2):
                merged += wor1[x]
            else:
                merged += word2[x]
        return merged
        
    mergeAlternately("word", "wort")


Solution.mergeAlternately("word", "wort")
